pkg/controller/controller_instance.go

```diff
@@ -2030,6 +2044,77 @@ func (c *controller) setServiceInstanceStartOrphanMitigation(toUpdate *v1beta1.S
 	)
 }
 
+// checkClassAndPlanForDeletion looks at serviceClass and servicePlan and
+// if either has been deleted, will block a new instance creation. If
+//
+func (c *controller) checkClassAndPlanForDeletion(instance *v1beta1.ServiceInstance, serviceClass *v1beta1.ClusterServiceClass, servicePlan *v1beta1.ClusterServicePlan) error {
```

Comment:
We're using the term "deleted" a lot, which is ambiguous, as the K8S resources haven't been deleted. What we're really trying to say is that they've been removed from the broker catalog. I'd rename this method to be `checkForRemovedClassAndPlan`.

Reply:
done